config/config.go

```diff
@@ -148,6 +149,17 @@ func endpointValidation(sl validator.StructLevel) {
 	}
 }
 
+// hasAPIKey searches through the provided endpoints to return whether or not
+// a given endpoint was supplied with an API Key.
+func hasAPIKey(endpointName provider.Name, endpoints []provider.Endpoint) bool {
+	for _, endpoint := range endpoints {
+		if endpoint.Name == endpointName && endpoint.APIKey != "" {
```

Comment:
nit: This feels hard coded to the `APIKey` entry. Could this be abstracted out where each provider has a list of required "provider_endpoints". It could help eliminate the really specific `if prov == provider.ProviderPolygon` below as well.

Reply:
I've made it a little cleaner by creating a custom bool type called `APIKeyRequired` and setting that as the value type in the `SupportedProviders` map. The `hasAPIKey` method is still required however since `cfg.endpoints` is a list so it needs to be iterated over to find the provider you're checking for.